### ISSUE TITLE
[WARN ] [e.internal.SseItemStatesEventBuilder] - Attempting to send a state update of an item which doesn’t exist: undefined

### ISSUE THREAD

**Issue by mstormi** [openhab/openhab-webui#2956]
This is a fairly frequently occuring warning, and often the origin/cause is not apparent

Trouble then is, it's hard to find out *what* is the item and which widget is it used in.

Please make that information part of the error message to help with debugging.

**Comment by rkoshak** [comment]
Unfortunately, I don't think that information is available to the server. All the server knows is *something* froim MainUI requested the state of an Item named `undefined` through the SSE. 

The code in question is at https://github.com/openhab/openhab-core/blob/1a91ef2b4427efda49c9f46d3658a751869ceced/bundles/org.openhab.core.io.rest.sse/src/main/java/org/openhab/core/io/rest/sse/internal/SseItemStatesEventBuilder.java#L84

All that has access to is the list of names of the Items MainUI has subscribed for events from. So to change that log and to make more information available there to add to the error log changes will need to be made to openhab-core first.

**Comment by stefan-hoehn** [comment]
so, @rkoshak, do you think the root cause is an issue in the Main UI (IMHO it is)?

**Comment by rkoshak** [comment]
The root cause is it's really hard to tell where this write is coming from as an end user. But if there can be a way to make the error more informative, it's going to involve changes to both MainUI and core to address it. 

I'm not certain it can be addressed but I'm no expert. 

MainUI could make tracking it down easier perhaps if it somehow logged to the browser console when an Item comes up `undefined` as the result of an expression and maybe give the name or location of the widget with the expression. Even just logging the expression could be enough if a clue to help track it down.

**Comment by florian-h05** [comment]
This issue can be fixed by making the UI not request the Item state if the Item name comes up as `undefined`, see #2965.
This is the only way of fixing this that came to my mind, but this fix unfortunately imposes the limitation that state tracking will not work for an Item named `undefined`. 
IMHO this limitation is very minor, I don't think one is going to name an Item `undefined`. WDYT?

**Comment by mstormi** [comment]
that limitation for sure is minor, but will this help with tracking down UI definition errors, i.e. which widget uses a wrong item name?
that was my intention here ...

**Comment by florian-h05** [comment]
No, this won’t help with that, but there is no reliable way to track down the faulting expression, so this is the best solution IMHO.
Function.caller is no part of the standard anymore so we shouldn’t use it.

The user will see a `-` in that case, and in most cases it might not even be a real issue, e.g. for some custom widgets.
